Firebase/Auth/Source/FIRAuth.m

```diff
@@ -790,25 +790,23 @@ - (void)signInWithCustomToken:(NSString *)token
   dispatch_async(FIRAuthGlobalWorkQueue(), ^{
     FIRAuthResultCallback decoratedCallback =
         [self signInFlowAuthResultCallbackByDecoratingCallback:completion];
-    FIRVerifyCustomTokenRequest *request =
-        [[FIRVerifyCustomTokenRequest alloc] initWithToken:token
-                                      requestConfiguration:_requestConfiguration];
-    [FIRAuthBackend verifyCustomToken:request
-                             callback:^(FIRVerifyCustomTokenResponse *_Nullable response,
-                                        NSError *_Nullable error) {
-      if (error) {
-        decoratedCallback(nil, error);
-        return;
-      }
-      [self completeSignInWithAccessToken:response.IDToken
-                accessTokenExpirationDate:response.approximateExpirationDate
-                             refreshToken:response.refreshToken
-                                anonymous:NO
-                                 callback:decoratedCallback];
+    [self internalSignInAndRetrieveDataWithCustomToken:token
+                                            completion:^(FIRAuthDataResult *_Nullable authResult,
+                                                         NSError * _Nullable error) {
+      decoratedCallback(authResult.user, error);
     }];
   });
 }
 
+- (void)signInAndRetrieveDataWithCustomToken:(NSString *)token
+                                  completion:(FIRAuthDataResultCallback)completion {
+  dispatch_async(FIRAuthGlobalWorkQueue(), ^{
+    FIRAuthDataResultCallback decoratedCallback =
+        [self signInFlowAuthDataResultCallbackByDecoratingCallback:completion];
+    [self internalSignInAndRetrieveDataWithCustomToken:token completion:decoratedCallback];
```

Comment:
Just double checking: why this method has non-nullable completion param and 'internalSignInAndRetrieveDataWithCustomToken:completion:' has nullable 'completion'?

BTW: 'signInWithCustomToken: completion:' has nullable param as well.

Reply:
In the header file this completion is nullable, the compiler somehow didn't complain about it being non-null here.  Done.